datafusion/expr/src/expr_schema.rs

```diff
@@ -173,7 +177,13 @@ impl ExprSchemable for Expr {
             | Expr::WindowFunction { .. }
             | Expr::AggregateFunction { .. }
             | Expr::AggregateUDF { .. } => Ok(true),
-            Expr::IsNull(_) | Expr::IsNotNull(_) | Expr::Exists(_) => Ok(false),
+            Expr::IsNull(_)
+            | Expr::IsNotNull(_)
+            | Expr::Exists { .. }
+            | Expr::InSubquery { .. } => Ok(false),
```

Comment:
I think a`IN` can be nullable if its exprs are nullable:

```sql
alamb=# select null IN (select * from a);
 ?column? 
----------
 
(1 row)
```

So maybe  `Expr::InSubquery` should return `expr.nullable(input_schema)` too

Reply:
Thanks. Fixed.